java/src/lib.rs

```diff
@@ -15,8 +16,93 @@ mod ffi_test;
 #[cfg(ffi_test)]
 pub use ffi_test::*;
 
+#[allow(dead_code)]
+/// Create a `JObject` from Rust's `String`. In case of an error
+/// this method throws Java exception and returns `JObject::null`
+fn create_glide_string_from_string<'local>(
+    env: &mut JNIEnv<'local>,
+    str: String,
+) -> JObject<'local> {
+    let Ok(bytearr) = env.byte_array_from_slice(str.as_bytes()) else {
+        let _ = env.throw("Failed to allocate byte array");
+        return JObject::null();
+    };
+
+    // TODO cache class
+    let Ok(class) = Desc::<JClass>::lookup("glide/api/models/GlideString", env) else {
+        let _ = env.throw("JNI: unable to find class GlideString");
+        return JObject::null();
+    };
+
+    // Call GlideString.of(byte[]) static method
+    let glide_string = match env.call_static_method(
+        class,
+        "of",
+        "([B)Lglide/api/models/GlideString;",
+        &[(&bytearr).into()],
+    ) {
+        Ok(glide_string) => glide_string,
+        Err(e) => {
+            let _ = env.throw(format!("JNI: failed to call GlideString.of. {:?}", e));
+            return JObject::null();
+        }
+    };
+
+    match JObject::try_from(glide_string) {
+        Ok(obj) => obj,
+        Err(e) => {
+            let _ = env.throw(format!("JNI: JObject::try_from failed!. {:?}", e));
+            JObject::null()
+        }
+    }
+}
+
+/// Create a `JObject` from Rust's `Vec<u8>`. In case of an error
+/// this method throws Java exception and returns `JObject::null`
+fn create_glide_string_from_bytes<'local>(
```

Comment:
You don't need this either if you return `byte[]` and construct `GlideString` by java means.
Futhermore, performance of JNI class/method lookup is terrible.

Reply:
Ack. We could do this (construct `GlideString` in Java) but the current code does check what is expected.
The most performant option we have - remove `GlideString` and use bytes directly, but in this case, we will need to keep the `String` API (i.e. using my initial proposal of using overload with `byte[]`)